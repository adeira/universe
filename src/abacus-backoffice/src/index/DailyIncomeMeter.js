// @flow

import { graphql, useFragment } from '@adeira/relay';
import { LayoutBlock, Meter, Money } from '@adeira/sx-design';
import type { Node } from 'react';

import refineSupportedCurrencies from '../refineSupportedCurrencies';
import type { DailyIncomeMeterFragment$key } from './__generated__/DailyIncomeMeterFragment.graphql';

type Props = {
  +data: DailyIncomeMeterFragment$key,
};

export default function DailyIncomeMeter(props: Props): Node {
  const dailyReport = useFragment(
    graphql`
      fragment DailyIncomeMeterFragment on AnalyticsDailyReportInfo {
        total {
          unitAmount
          unitAmountCurrency
        }
      }
    `,
    props.data,
  );

  const dailyGoal = 2500; // TODO: MXN only

  return (
    <LayoutBlock>
      <Meter
        min={0}
        max={dailyGoal}
        value={
          dailyReport.total.unitAmount / 100 // adjusted for centavo
        }
        low={dailyGoal * 0.8}
        optimum={dailyGoal}
      />
      <Money
        priceUnitAmount={
          dailyReport.total.unitAmount / 100 // adjusted for centavo
        }
        priceUnitAmountCurrency={refineSupportedCurrencies(dailyReport.total.unitAmountCurrency)}
      />
    </LayoutBlock>
  );
}
