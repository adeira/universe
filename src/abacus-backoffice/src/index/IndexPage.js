// @flow

import * as React from 'react';
import { Note } from '@adeira/sx-design';
import { graphql, useLazyLoadQuery } from '@adeira/relay';

import type { IndexPageQuery } from './__generated__/IndexPageQuery.graphql';

export default function IndexPage(): React.Node {
  const data = useLazyLoadQuery<IndexPageQuery>(
    /* eslint-disable relay/unused-fields */
    graphql`
      query IndexPageQuery {
        whoami {
          id
          humanReadableType
        }
      }
    `,
    /* eslint-enable relay/unused-fields */
  );

  return (
    <>
      <Note tint="warning">work in progress</Note>
      <pre>{JSON.stringify(data, null, 2)}</pre>
    </>
  );
}
