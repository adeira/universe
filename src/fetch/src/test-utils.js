// @flow

/* eslint-disable import/no-extraneous-dependencies */

import { setupServer, type Server } from 'msw/node';

export { rest } from 'msw';

export const server: Server = setupServer();
