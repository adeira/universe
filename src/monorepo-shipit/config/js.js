// @flow strict

import type { ConfigType } from '../ConfigType.flow';

module.exports = ({
  getStaticConfig() {
    return {
      repository: 'git@github.com:adeira/js.git',
    };
  },
  getPathMappings() {
    return new Map([['src/js/', '']]);
  },
}: ConfigType);
