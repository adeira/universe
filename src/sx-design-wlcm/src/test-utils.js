// @flow

import { init as FbtInit, IntlVariations as FbtIntlVariations } from 'fbt';
import { render as renderWithoutProviders } from '@testing-library/react';

const initFbt = () => {
  FbtInit({
    translations: {}, // TODO
    hooks: {
      getViewerContext: () => ({
        GENDER: FbtIntlVariations.GENDER_UNKNOWN,
        regionalLocale: 'en_US',
        locale: 'en_US',
      }),
    },
  });
};

// export custom render method(s) and other helpers:
export { initFbt, renderWithoutProviders as render };
