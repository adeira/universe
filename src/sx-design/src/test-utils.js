// @flow

import { type Element } from 'react';
import { init as FbtInit, IntlVariations as FbtIntlVariations } from 'fbt';
// eslint-disable-next-line import/no-extraneous-dependencies
import {
  render as renderWithoutProviders,
  type RenderOptionsWithoutCustomQueries,
} from '@testing-library/react';

import getFbtTranslationsForLocale from './getFbtTranslationsForLocale';
import type { SupportedLocales } from './constants';

const initFbt = () => {
  FbtInit({
    translations: getFbtTranslationsForLocale('en-US'),
    hooks: {
      getViewerContext: () => ({
        GENDER: FbtIntlVariations.GENDER_UNKNOWN,
        locale: 'en-US',
      }),
    },
  });
};

const renderWithProviders = (
  ui: Element<any>,
  {
    locale = 'en-US',
    ...renderOptions
  }: { +locale: SupportedLocales, ...RenderOptionsWithoutCustomQueries } = {},
): $FlowFixMe => {
  return renderWithoutProviders(ui, {
    wrapper: ({ children }) => {
      const SxDesignProvider = require('./SxDesignProvider').default;
      return <SxDesignProvider locale={locale}>{children}</SxDesignProvider>;
    },
    ...renderOptions,
  });
};

// re-export everything
// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';

// override render method
export { initFbt, renderWithProviders as render, renderWithoutProviders };
