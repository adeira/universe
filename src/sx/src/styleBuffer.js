// @flow strict

type GenericStyleBuffer = {|
  +className: string,
  +styleName: string,
  +styleValue: string,
  +pseudo?: string,
|};

type Hash = string;

// Here we are collecting all the styles while doing SSR.
export const styleBuffer = (new Map(): Map<Hash, GenericStyleBuffer>);

// Here we are collecting all the styles while doing SSR for media queries.
export const mediaStyleBuffer = (new Map(): Map<
  string, // normalized media query
  Map<Hash, GenericStyleBuffer>,
>);
