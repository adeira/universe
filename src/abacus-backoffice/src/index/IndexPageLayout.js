// @flow

import { Heading } from '@adeira/sx-design';
import fbt from 'fbt';
import * as React from 'react';

import Layout from '../Layout';
import LayoutHeading from '../LayoutHeading';
import IndexPage from './IndexPage';

export default function IndexPageLayout(): React.Node {
  return (
    <Layout>
      <LayoutHeading
        heading={
          <Heading>
            <fbt desc="quick analytics (stats) on the Abacus homepage">Quick analytics</fbt>
          </Heading>
        }
        description={<fbt desc="have a great day message">Have a great day!</fbt>}
      />

      <IndexPage />
    </Layout>
  );
}
