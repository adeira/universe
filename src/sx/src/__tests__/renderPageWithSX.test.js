// @flow

import path from 'path';
import prettier from 'prettier';
import { generateTestsFromFixtures } from '@adeira/test-utils';
import { sprintf } from '@adeira/js';
import TestRenderer from 'react-test-renderer';

import * as sx from '../../index';
import { styleBuffer, mediaStyleBuffer } from '../styleBuffer';

beforeEach(() => {
  styleBuffer.clear();
  mediaStyleBuffer.clear();
});

const renderPageMock = () => ({
  html: '',
  head: [''],
  styles: [''],
});

generateTestsFromFixtures(path.join(__dirname, 'fixtures'), (input) => {
  const stylesheetsDefinition = JSON.parse(input);
  const styles = sx.create(stylesheetsDefinition);

  let output = '';

  // 1) print final atomic CSS
  const renderer = TestRenderer.create(sx.renderPageWithSX(renderPageMock).styles[0]);
  const css = renderer.root.children[0].toString();
  output += prettier.format(css, { filepath: 'test.css' });

  output += '\n~~~~~~~~~~ USAGE ~~~~~~~~~~\n';

  // 2) print CSS classes when calling the stylesheet name
  Object.keys(stylesheetsDefinition).forEach((stylesheetName) => {
    output += sprintf(
      `
className={styles('%s')}
  â†“ â†“ â†“
class="%s"
`,
      stylesheetName,
      styles(stylesheetName),
    );
  });

  return output;
});

it('works as expected', () => {
  const styles = sx.create({
    red: { color: 'red' },
    blue: { color: 'blue' },
    pseudo: {
      'color': 'green',
      ':hover': {
        color: 'red',
        textDecoration: 'underline',
      },
      ':focus': {
        color: 'purple',
      },
      '::after': {
        content: 'ðŸ¤“',
      },
    },
  });

  expect(styles('red')).toMatchInlineSnapshot(`"wUqnh"`);
  expect(styles('blue')).toMatchInlineSnapshot(`"_4fo5TC"`);

  expect(styles('red', 'blue')).toMatchInlineSnapshot(`"_4fo5TC"`); // blue wins
  expect(styles('blue', 'red')).toMatchInlineSnapshot(`"wUqnh"`); // red wins

  expect(styles('pseudo')).toMatchInlineSnapshot(`"PJDYD _4sFdkU _22QzO9 _3stS2V _14RYUP"`);
  expect(styles('pseudo', 'red')).toMatchInlineSnapshot(`"wUqnh _4sFdkU _22QzO9 _3stS2V _14RYUP"`); // red wins (non-hover)
});
