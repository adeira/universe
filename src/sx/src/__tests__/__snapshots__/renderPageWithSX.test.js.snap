// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: @supports.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "test": {
    "@supports (display: grid)": {
      "display": "grid"
    },
    "@supports not (display: grid)": {
      "float": "right"
    }
  },
  "testComplex": {
    "@supports (display: table-cell) and ((display: list-item) and (display:run-in))": {
      "color": "red"
    },
    "@supports (transform-style: preserve) or (-moz-transform-style: preserve) or (-o-transform-style: preserve) or (-webkit-transform-style: preserve)": {
      "color": "red"
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
@supports (display: grid) {
  ._3GKV06._3GKV06 {
    display: grid;
  }
}
@supports not (display: grid) {
  ._1cjV2G._1cjV2G {
    float: right;
  }
}
@supports (display: table-cell) and ((display: list-item) and (display: run-in)) {
  ._3DO79H._3DO79H {
    color: #f00;
  }
}
@supports (transform-style: preserve) or (-moz-transform-style: preserve) or
  (-o-transform-style: preserve) or (-webkit-transform-style: preserve) {
  ._4mzMj8._4mzMj8 {
    color: #f00;
  }
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('test')}
  ↓ ↓ ↓
class="_3GKV06 _1cjV2G"

className={styles('testComplex')}
  ↓ ↓ ↓
class="_3DO79H _4mzMj8"

`;

exports[`matches expected output: @supports-nested.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "test": {
    "@supports (display: grid)": {
      "color": "red",
      "@supports not (display: grid)": {
        "color": "blue",
        "@supports (display: table-cell) and ((display: list-item) and (display:run-in))": {
          "color": "orange"
        }
      }
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
@supports (display: grid) {
  ._3LkwQx._3LkwQx {
    color: #f00;
  }
  @supports not (display: grid) {
    ._3KVzHX._3KVzHX {
      color: #00f;
    }
    @supports (display: table-cell) and
      ((display: list-item) and (display: run-in)) {
      ._39dw1H._39dw1H {
        color: #ffa500;
      }
    }
  }
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('test')}
  ↓ ↓ ↓
class="_3LkwQx _3KVzHX _39dw1H"

`;

exports[`matches expected output: @unknown.error.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "test": {
    "@unknown": {
      "color": "red"
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
THROWN EXCEPTION:

Invariant Violation: Unsupported rule "@unknown"
`;

exports[`matches expected output: media-queries.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "test": {
    "color": "red",
    "@media print": {
      "color": "red"
    },
    "@media (min-width: 30em) and (max-width: 50em)": {
      "color": "blue"
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
.wUqnh {
  color: #f00;
}
@media print {
  .zIzjk.zIzjk {
    color: #f00;
  }
}
@media (min-width: 30em) and (max-width: 50em) {
  ._27Xk5F._27Xk5F {
    color: #00f;
  }
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('test')}
  ↓ ↓ ↓
class="wUqnh zIzjk _27Xk5F"

`;

exports[`matches expected output: media-queries-multiple.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "aaa": {
    "color": "red",
    "fontSize": 32,
    "@media print": {
      "color": "red",
      "fontSize": 16
    },
    "@media (min-width: 30em) and (max-width: 50em)": {
      "color": "blue"
    }
  },
  "bbb": {
    "@media print": {
      "color": "green"
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
.wUqnh {
  color: #f00;
}
._2ZJp7Y {
  font-size: 2rem;
}
@media print {
  .zIzjk.zIzjk {
    color: #f00;
  }
  .uxyoC.uxyoC {
    font-size: 1rem;
  }
  .v2909.v2909 {
    color: #008000;
  }
}
@media (min-width: 30em) and (max-width: 50em) {
  ._27Xk5F._27Xk5F {
    color: #00f;
  }
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('aaa')}
  ↓ ↓ ↓
class="wUqnh _2ZJp7Y zIzjk uxyoC _27Xk5F"

className={styles('bbb')}
  ↓ ↓ ↓
class="v2909"

`;

exports[`matches expected output: media-queries-nested.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "test": {
    "@media print": {
      "color": "red",
      "@media (max-width: 12cm)": {
        "color": "blue"
      }
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
@media print {
  .zIzjk.zIzjk {
    color: #f00;
  }
  @media (max-width: 12cm) {
    .Uxdbe.Uxdbe {
      color: #00f;
    }
  }
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('test')}
  ↓ ↓ ↓
class="zIzjk Uxdbe"

`;

exports[`matches expected output: media-queries-nested-deep.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "test": {
    "@media (max-width: 10cm)": {
      "color": "#F00000",
      "backgroundColor": "#FFFFFF",
      "@media (max-width: 9cm)": {
        "color": "#0F0000",
        "backgroundColor": "#FFFFFF",
        "@media (max-width: 8cm)": {
          "color": "#00F000",
          "backgroundColor": "#FFFFFF",
          "@media (max-width: 7cm)": {
            "color": "#000F00",
            "backgroundColor": "#FFFFFF",
            "@media (max-width: 6cm)": {
              "color": "#0000F0",
              "backgroundColor": "#FFFFFF",
              "@media (max-width: 5cm)": {
                "color": "#00000F",
                "backgroundColor": "#FFFFFF"
              }
            }
          }
        }
      }
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
@media (max-width: 10cm) {
  ._4bGnnn._4bGnnn {
    color: #f00000;
  }
  ._1kQaJH._1kQaJH {
    background-color: #fff;
  }
  @media (max-width: 9cm) {
    ._21BNws._21BNws {
      color: #0f0000;
    }
    ._3Lm6Uw._3Lm6Uw {
      background-color: #fff;
    }
    @media (max-width: 8cm) {
      ._2sV160._2sV160 {
        color: #00f000;
      }
      ._1bLvdm._1bLvdm {
        background-color: #fff;
      }
      @media (max-width: 7cm) {
        ._3TWByw._3TWByw {
          color: #000f00;
        }
        ._1adr9y._1adr9y {
          background-color: #fff;
        }
        @media (max-width: 6cm) {
          ._1EWUxi._1EWUxi {
            color: #0000f0;
          }
          ._2g7PIB._2g7PIB {
            background-color: #fff;
          }
          @media (max-width: 5cm) {
            ._3l858s._3l858s {
              color: #00000f;
            }
            ._22f0H3._22f0H3 {
              background-color: #fff;
            }
          }
        }
      }
    }
  }
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('test')}
  ↓ ↓ ↓
class="_4bGnnn _1kQaJH _21BNws _3Lm6Uw _2sV160 _1bLvdm _3TWByw _1adr9y _1EWUxi _2g7PIB _3l858s _22f0H3"

`;

exports[`matches expected output: media-queries-nested-pseudo.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "media": {
    "color": "red",
    "@media print": {
      "color": "red",
      ":hover": {
        "color": "pink"
      }
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
.wUqnh {
  color: #f00;
}
@media print {
  .zIzjk.zIzjk {
    color: #f00;
  }
  ._4rAdwD._4rAdwD:hover {
    color: #ffc0cb;
  }
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('media')}
  ↓ ↓ ↓
class="wUqnh zIzjk _4rAdwD"

`;

exports[`matches expected output: media-queries-same-style.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "style1": {
    "@media (min-width: 1280px)": {
      "color": "red"
    },
    "color": "blue"
  },
  "style2": {
    "color": "red"
  }
}
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
@media (min-width: 1280px) {
  ._3ZCUJ6._3ZCUJ6 {
    color: #f00;
  }
}
._4fo5TC {
  color: #00f;
}
.wUqnh {
  color: #f00;
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('style1')}
  ↓ ↓ ↓
class="_3ZCUJ6 _4fo5TC"

className={styles('style2')}
  ↓ ↓ ↓
class="wUqnh"

`;

exports[`matches expected output: pseudo-classes.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "link": {
    "textDecoration": "none",
    ":hover": {
      "textDecoration": "underline"
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
._134wwt {
  text-decoration: none;
}
._22QzO9:hover {
  text-decoration: underline;
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('link')}
  ↓ ↓ ↓
class="_134wwt _22QzO9"

`;

exports[`matches expected output: pseudo-classes-multiple.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "aaa": {
    "textDecoration": "none",
    ":hover": {
      "textDecoration": "none",
      "color": "red"
    },
    ":active": {
      "color": "red"
    }
  },
  "bbb": {
    ":hover": {
      "color": "blue"
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
._134wwt {
  text-decoration: none;
}
.XJjf5:hover {
  text-decoration: none;
}
._4sFdkU:hover {
  color: #f00;
}
.UddGc:hover {
  color: #00f;
}
.ERwS0:active {
  color: #f00;
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('aaa')}
  ↓ ↓ ↓
class="_134wwt XJjf5 _4sFdkU ERwS0"

className={styles('bbb')}
  ↓ ↓ ↓
class="UddGc"

`;

exports[`matches expected output: pseudo-elements.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "link": {
    "textDecoration": "none",
    ":hover": {
      "textDecoration": "underline"
    },
    "::after": {
      "content": "∞"
    },
    "::before": {
      "content": "∞"
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
._134wwt {
  text-decoration: none;
}
._22QzO9:hover {
  text-decoration: underline;
}
._2Rpe4g::after {
  content: "∞";
}
._3ou8wp::before {
  content: "∞";
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('link')}
  ↓ ↓ ↓
class="_134wwt _22QzO9 _2Rpe4g _3ou8wp"

`;

exports[`matches expected output: simple-stylesheet.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "redStyle": {
    "color": "red"
  },
  "blueStyle": {
    "color": "blue"
  },
  "greenStyle": {
    "color": "green"
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
.wUqnh {
  color: #f00;
}
._4fo5TC {
  color: #00f;
}
.PJDYD {
  color: #008000;
}

~~~~~~~~~~ USAGE ~~~~~~~~~~

className={styles('redStyle')}
  ↓ ↓ ↓
class="wUqnh"

className={styles('blueStyle')}
  ↓ ↓ ↓
class="_4fo5TC"

className={styles('greenStyle')}
  ↓ ↓ ↓
class="PJDYD"

`;
